Sources/Target/Linux/Thread.cpp

```diff
@@ -164,6 +164,40 @@ ErrorCode Thread::writeCPUState(Architecture::CPUState const &state) {
       ProcessThreadId(process()->pid(), tid()), info, state);
 }
 
+void Thread::fillStoppointData() {
+  HardwareBreakpointManager *hw_bpm = process()->hardwareBreakpointManager();
+
+  if (hw_bpm) {
+    Address addr;
+    BreakpointManager::Mode mode;
+
+    if (hw_bpm->wasHit(addr, mode)) {
+      if (mode != BreakpointManager::kModeExec) {
+        _stopInfo.watchAddr = addr;
+      }
+
+      switch (static_cast<int>(mode)) {
+      case BreakpointManager::kModeExec:
+        _stopInfo.reason = StopInfo::kReasonBreakpoint;
+        return;
+      case BreakpointManager::kModeWrite:
+        _stopInfo.reason = StopInfo::kReasonWriteWatchpoint;
+        return;
+      case BreakpointManager::kModeRead:
+        _stopInfo.reason = StopInfo::kReasonReadWatchpoint;
+        return;
+      case BreakpointManager::kModeRead | BreakpointManager::kModeWrite:
```

Comment:
Haha, I see. Annoying compiler :)

On Sat, Apr 9, 2016 at 11:50 AM, Francis Ricci <notifications@github.com>
wrote:

> In Sources/Target/Linux/Thread.cpp
> <https://github.com/facebook/ds2/pull/243#discussion_r59119532>:
>
> > +    if (hw_bpm->wasHit(addr, mode)) {
> > +      if (mode != BreakpointManager::kModeExec) {
> > +        _stopInfo.watchAddr = addr;
> > +      }
> > +
> > +      switch (static_cast<int>(mode)) {
> > +      case BreakpointManager::kModeExec:
> > +        _stopInfo.reason = StopInfo::kReasonBreakpoint;
> > +        return;
> > +      case BreakpointManager::kModeWrite:
> > +        _stopInfo.reason = StopInfo::kReasonWriteWatchpoint;
> > +        return;
> > +      case BreakpointManager::kModeRead:
> > +        _stopInfo.reason = StopInfo::kReasonReadWatchpoint;
> > +        return;
> > +      case BreakpointManager::kModeRead | BreakpointManager::kModeWrite:
>
> This doesn't work - error: case value ‘6’ not in enumerated type
> ‘ds2::BreakpointManager::Mode’ [-Werror=switch]
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly or view it on GitHub
> <https://github.com/facebook/ds2/pull/243/files/12c7a36ec464211fdf5b11c22f29a9e8f7bb4e11..d875a486968a22f7eab2de9b787315cc6ca31fde#r59119532>
>



-- 
Stephane Sezer